packages/garbo/src/resources/candyMap.ts

```diff
@@ -0,0 +1,124 @@
+import { AcquireItem, Outfit } from "grimoire-kolmafia";
+import {
+  $familiar,
+  $item,
+  get,
+  getSaleValue,
+  have,
+  sum,
+} from "libram";
+import {
+  canEquip,
+  getOutfits,
+  outfitPieces,
+  outfitTreats,
+  print,
+  runChoice,
+  toItem,
+  use,
+  visitUrl,
+} from "kolmafia";
+import { GarboTask } from "../tasks/engine";
+
+const HOUSE_NUMBERS = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
+let treated = false;
+
+let blockHtml = "";
+
+function getBlockHtml(): string {
+  blockHtml ||= visitUrl("place.php?whichplace=town&action=town_trickortreat");
+  return blockHtml;
+}
+
+function treatValue(outfit: string): number {
+  return sum(
+    Object.entries(outfitTreats(outfit)),
+    ([candyName, probability]) => probability * getSaleValue(toItem(candyName))
+  );
+}
+
+export function getTreatOutfit(): string {
+  if (!get("freecandy_treatOutfit")) {
+    const availableOutfits = getOutfits().filter((name) =>
+      outfitPieces(name).every((piece) => canEquip(piece))
+    );
+
+
+    if (!availableOutfits.length) {
+      print("You don't seem to actually have any outfits available, my friend!");
+    }
+
+  }
+
+  return get("freecandy_treatOutfit");
+}
+
+export function treatOutfit(): Outfit {
+  const outfit = new Outfit();
+  const pieces = outfitPieces(getTreatOutfit());
+  for (const piece of pieces) {
+    if (!outfit.equip(piece))
+      print(`Could not equip all pieces of treat outfit: aborted on ${piece}`);
+  }
+
+  outfit.equip($item`lucky Crimbo tiki necklace`);
+  outfit.equip($familiar`Trick-or-Treating Tot`);
+  return outfit;
+}
+
+export function freeCandyValue(): number {
+    const outfitCandyValue = treatValue(getTreatOutfit());
+    const totOutfitCandyMultiplier = have($familiar`Trick-or-Treating Tot`) ? 1.6 : 1;
+    const bowlValue = (1 / 5) * getSaleValue($item`huge bowl of candy`);
+    const prunetsValue = have($familiar`Trick-or-Treating Tot`)
+      ? 4 * 0.2 * getSaleValue($item`Prunets`)
+      : 0;
+
+    const outfitCandyTotal = 3 * outfitCandyValue * totOutfitCandyMultiplier;
+    let _baseAdventureValue = (1 / 5) * (outfitCandyTotal + bowlValue + prunetsValue);
+    return _baseAdventureValue;
+}
+
+function shouldAcquireCandyMap(): boolean {
+  if(freeCandyValue() < getSaleValue($item`map to a candy-rich block`))
+    return true;
+  else
+    return false;
+}
+
+function useCandyMapTask(): GarboTask {
+  return {
+  name: "Acquire Candy Map",
+  completed: () => shouldAcquireCandyMap(),
+  do: (): void => {
+    AcquireItem($item`map to a candy-rich block`)
+  },
+};
+}
+
+function doCandyTreat(): GarboTask {
+  return {
+    name: "Treat",
+    ready: () => !treated,
+    completed: () => !have($item`map to a candy-rich block`) || get("_mapToACandyRichBlockUsed") 
+    || !getBlockHtml().match(/whichhouse=\d*>[^>]*?house_l/),
```

Comment:
remove the item check, because we're now using the item as part of the other task; move the pref check to the `ready` condition. I think we can eschew all of the specialized getBlockHtml() architecture because we aren't going to more than one block.

Reply:
How will we determine we've hit every house in the block?